### ISSUE TITLE
[Feature request]: Keyboard shortcuts to select a save slot

### ISSUE THREAD

**Issue by ReyVGM** [visualboyadvance-m/visualboyadvance-m#1208]
### Request box

Please add keyboard shortcuts to select specific save slots. Right now you can only save to a specific slot and load from that specific slot.

### Reason box

It makes save/loading easier and reduces the amount of shortcuts needed for multiple slots.

It's easier, for example (it can be any key), to use numbers 0-9 to select between 10 slots, then use F5 to save to that slot and F7 to load from that slot. Of course, the current shortcuts can stay for those who are used to them.

Here are some examples from other emulators (snes9x & FCEUX)

### Examples

![image](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/7338867/343c9772-e7f1-456a-92dd-627f711d8313)
![image](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/7338867/017ea8eb-fed0-474a-a507-fec813a9e839)
![image](https://github.com/visualboyadvance-m/visualboyadvance-m/assets/7338867/b4fc1f6d-103e-485b-8162-d62195f3405c)

**Comment by Squall-Leonhart** [comment]
We already have hotkeys for saving to and loading from specific slots, we also have an undefined hotkey support for enabling slot incrementation.

**Comment by ReyVGM** [comment]
Sorry for reopening this, but what I'm requesting is different from what you mentioned.

If what I'm asking is not something the team wants to implement because the current options are good enough, then I would understand the decision. But I just want to make sure that you're clear that I'm asking for something different and more like how it works on most other emulators for the NES and SNES.

**Comment by Steelskin** [comment]
The way to do it would be to add options for "Select save state [0-9]" as we already have shortcuts for save/load current state slot

I think the main issue is that right now, command options are tied to the menu, so we'd either have to add a menu or refactor the way we handle shortcuts altogether. I do think it is worth implementing but some more work needs to happen before we can get there.

**Comment by ReyVGM** [comment]
Thanks for the reply Steelskin. By the way, it wouldn't be "Select save state [0-9]" but rather "Select save SLOT [0-9]".

Once you have a slot selected, then you would use a single key to save and a single key to load.

I don't know if that's what you meant, but I felt the need to clarify it even further.

**Comment by falcie1337** [comment]
yeah i agree with this, it's a clean way to manage your save states
